### ISSUE TITLE
When Nullable is enabled, Roslyn incorrectly warns of possible null dereference when MemberNotNullWhenAttribute is used with logical or operators

### ISSUE THREAD

**Issue by Dancpaz** [dotnet/roslyn#53255]
Visual Studio 2019

The problem appears in the following code, in a Blazor Server project:
```
        public void RemoveItem(SomeItem item)
        {
            if (!IsPhaseOneLoaded | !IsPhaseTwoLoaded) throw new InvalidOperationException();
            if (!FirstCollection.Remove(item))
                throw new InvalidOperationException();

            SecondCollection.Add(item);
            IsDirty = true;
        }
```

Roslyn warns that FirstCollection and SecondCollection may be null, but IsPhaseOneLoaded and IsPhaseTwoLoaded both have the appropriate MemberNotNullWhenAttribute applied.  Changing the logical or operator to a conditional logical or operator causes the warnings to disappear.

This is what I will do for now, but Roslyn should understand logical or in the context of nullability.
